internal/middlewares/movies.go

```diff
@@ -0,0 +1,66 @@
+package middlewares
+
+import (
+	"context"
+	"net/http"
+
+	".com/internal/db"
+	".com/internal/models"
+	".com/internal/server"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/render"
+	"github.com/google/uuid"
+)
+
+// Define a custom type for the context key
+type movieContextKeyType struct{}
+
+// Implement the String method for the context key type
+func (k *movieContextKeyType) String() string {
+	return "movie"
+}
+
+// Create a global instance of the movie context key
+var movieContextKey = &movieContextKeyType{}
```

Comment:
is this needed ? i've checked some source code of 

`return ctx.Value(movieContextKeyType{}).(*models.Movie)`

`return context.WithValue(ctx, movieContextKeyType{}, movie)`